Raw YAHPO Benchmarks

PhMueller opened the pull request:
Interface to evaluate the true performance of a configuration (not as surrogate or tabular benchmark)

PhMueller committed:
Init Commit

PhMueller committed:
Init Commit

PhMueller committed:
First working version (SO)

PhMueller committed:
First working version (SO)

PhMueller committed:
First working version (SO)

PhMueller committed:
First working version (SO)

PhMueller committed:
Change evaluation call wrt used learner.

PhMueller committed:
Change evaluation call wrt used learner.

PhMueller committed:
Add MO support

PhMueller committed:
Add MO support

KEggensperger commented on extra_requirements/rbv2.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "rbv2": ["yahpo_gym@git+https://github.com/pfistfl/yahpo_gym#egg=yahpo_gym&subdirectory=yahpo_gym", "rpy2", "openml==0.10.2"],
+  "rbv2_old": ["yahpo_gym@git+https://github.com/pfistfl/yahpo_gym#egg=yahpo_gym&subdirectory=yahpo_gym", "rpy2", "numpy", "configspace", "pandas==1.2.4","scikit-learn==0.24.2", "openml==0.10.2"]
```
there are missing dependency versions

KEggensperger commented on hpobench/container/recipes/ml/Singularity.rbv2Benchmark:
```diff
@@ -0,0 +1,65 @@
+    cd /home \
+    && git clone https://github.com/pfistfl/HPOBench.git \
+    && cd HPOBench \
+    && echo "Please never push a recipe that checks out any other branch than development or master" \
```
This contradicts the next line where you checkout a different branch ...

PhMueller committed:
Debug push

PhMueller committed:
Debug push

PhMueller committed:
Add interface to IAML package

PhMueller committed:
Add interface to IAML package

PhMueller committed:
Update Recipe

PhMueller committed:
Update Recipe

PhMueller committed:
Clean Up Benchmark

PhMueller committed:
Clean Up Benchmark

PhMueller committed:
Add automatic data downloading

PhMueller committed:
Add automatic data downloading

PhMueller committed:
Rename rbv2 -> yahpo_benchmark

PhMueller committed:
Rename rbv2 -> yahpo_benchmark

PhMueller committed:
Remove Flake error

PhMueller committed:
Remove Flake error

PhMueller committed:
Merge branch 'development' into yahpo_raw

PhMueller committed:
update test

PhMueller committed:
update test

PhMueller committed:
Fix data manager

PhMueller committed:
Fix data manager

ayushi-3536 committed:
Benchmark: Fair Adult from MO-ASHA (#148)

We add the benchmark from the MO-ASHA paper by Schmucker et al.

It is a MO benchmark, training an MLP on the Adult data set.

ayushi-3536 committed:
Multi Objective CNN benchmark: Flowers and Fashion (#147)

Added mo cnn benchmarks from bag of baseline paper

We deviate from the original benchmark in two points: 
* we return as cost only the  training time instead of the total elapsed time
* we return as objective for minimization instead of `-100 * accuracy` now `1 - accuracy` to achieve better output scalings. 

Co-authored-by: ayushi-3536 <ayushi-3536@github.com>
Co-authored-by: Philipp Müller <muller-phil@gmx.net>

LGro commented on hpobench/util/data_manager.py:
```diff
@@ -980,3 +981,36 @@ def _load_json(path):
+            import git
+            git.Repo.clone_from(url='https://github.com/pfistfl/yahpo_data.git',
+                                to_path=str(self.data_dir), branch='main', multi_options=['--depth 1'])
+            self.logger.info(f'Successfully cloned data from repo to {self.data_dir}')
```
Given that `yahpo_data` is licensed under [AGPL](https://github.com/slds-lmu/yahpo_data/blob/main/LICENSE), how do you judge the compatibility with this repo's more permissive Apache-2.0 license, @KEggensperger and @PhMueller?

LGro replied:
And the maintainers over there just [relicensed to Apache-2.0](https://github.com/slds-lmu/yahpo_data/commit/f9b8cd24fd8e7d9cc8fb7372f9640128941255e5). My question is resolved then, I think :relaxed:

PhMueller committed:
HPOBench with python >= 3.10 (#162)

Make installable for Python 3.10.6
* Add 3.10 to test pipeline
* Remove upper Python version limit
* Change python-versions to strings
* Surround action python-version with quotation marks

* Install paramnet dependencies only for Python <= 3.9
* Disable local paramnet test for Python > 3.9

Co-authored-by: Dominik Woiwode <dwoiwode@users.noreply.github.com>

PhMueller committed:
Speed up CI testing (#163)

* Speed up tests by disabling slow tests.

PhMueller committed:
Refactor PR (#164)

* Refactor: Rename AbstractBenchmark Class

PhMueller committed:
Nas201 - v0.0.6 (#165)

* Update 201 version 0.0.6

- scale the misclassification rate form 0 to 100 -> 0 and 1
- Update test cases
- link to the correct container version
- Fix Nas201 Test Cases
- remove a Github action trigger. It started all test cases twice. 
- Fix GithubActions: Install correct xgboost version for 3.10

PhMueller committed:
Nas101 + Nas1shot1 v0.0.5 - Multi Objective (#166)

* Nas101 v.0.0.5
* Nas1shot1 v.0.0.5
* Test Cases for 101

PhMueller committed:
MO CNN and MO Adult v0.0.2 (#167)

Update to 0.0.2

PhMueller committed:
Yahpo Raw

PhMueller committed:
Yahpo Raw

PhMueller committed:
YAHPO Surrogate V0.0.2

PhMueller committed:
YAHPO Surrogate V0.0.2

PhMueller committed:
Add Recipe for Yahpo raw benchmakr

PhMueller committed:
Add Recipe for Yahpo raw benchmakr

PhMueller committed:
refactor

PhMueller committed:
Merge branch 'yahpo_raw' of https://github.com/PhMueller/HPOBench into yahpo_raw

PhMueller committed:
Fix Test file
